src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -281,9 +282,9 @@ public boolean getForceRedstoneUpdate()
     @HasResult
     public static class CreateFluidSourceEvent extends BlockEvent
     {
-        public CreateFluidSourceEvent(World world, BlockPos pos, BlockState state)
+        public CreateFluidSourceEvent(IWorldReader world, BlockPos pos, BlockState state)
         {
-            super(world, pos, state);
+            super((IWorld) world, pos, state);
```

Comment:
You can't just blindly cast things.

Reply:
What do you think about it? `super(world instanceof IWorld ? (IWorld) world : null, pos, state);`